### DISCUSSION TITLE
Total Count

### DISCUSSION THREAD

**Discussion by dchilton6** [syssi/esphome-total-count#15] [Q&A]
I have setup esphome-total-count to monitor a momentary contact on a tipping rain gauge. it reliably counts the bucket tips and provides a running total. 
I use a helper utility meter to provide daily, weekly, monthly, yearly totals.
Two questions:
How do I multiply the count by 0.203 to provide the mm of rain?
The total count sets itself up as a number.total.count but the helper needs a sensor.
Any ideas?

**Comment by syssi** [comment]
You have to use a template sensor which picks the state of the counter and does the multiplication. Please give it a try. I will give some guidance if you have trouble to set it up.

**Reply by dchilton6** [reply]
Thanks for the quick reply. 
On holiday this week so not too easy to access configs!
I did have a template setup for a n earlier version of the rain gauge which had an on and off so had to add the two together and when it reset at midnight it added 1 on.
This version is off and just makes a momentary on as it tips which total count is working reliably. 
I also wanted it setup in earphone rather that configuration.yaml

**Reply by syssi** [reply]
Just for clarification: I'm talking of a template sensor located at the configuration YAML of the ESPHome node. Something like this:

```
sensor:
  - platform: total_count
    total_count:
      id: my_total_count
      name: "${name} total count"

  - platform: template
    name: "${name} total count adjusted"
    lambda: |-
      if (id(my_total_count).state) {
        return id(my_total_count).state * 0.203f;
      } else {
        return NAN;
      }
    update_interval: 60s
```

**Comment by dchilton6** [comment]
Thanks again for your help. 
I have had a number of different setups and configs for this and all have had a slight issue or unreliable.
This seemed a perfect fit. 
I will let you know how it goes once I get home and can test out in situ .
FYI:
I am gradually building up a weather station. 
Next step is 3D printing parts for wind speed.
Again this seems a perfect fit for counting the contacts on each revolution for a time period then calculating average speed.
I presume I can use it on the same esp32 on another gpio with any conflict?

**Reply by syssi** [reply]
That's correct. You can instantiate the component multiple times. See

https://github.com/syssi/esphome-total-count/blob/main/esp32-example-advanced.yaml
